src/lib.rs

```diff
@@ -209,6 +330,53 @@ pub struct Group {
     pub satisfied_by: Option<String>,
 }
 
+/// A signature describes a cryptographic signature of the parcel list.
+///
+/// In the current implementation, a signature signs the list of parcels that belong on
+/// an invoice. The signature, in the current implementation, is an Ed25519 signature
+/// and is signed by the private counterpart of the given public key.
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(deny_unknown_fields, rename_all = "camelCase")]
+pub struct Signature {
+    // The cleartext name of the user who signed
+    pub by: String,
+    // The signature block, encoded as hex chars
+    pub signature: String,
+    // The public key, encoded as hex chars
+    pub key: String,
+    pub role: SignatureRole,
+}
+
+#[derive(Error, Debug)]
+pub enum SignatureError {
+    #[error("not all signatures can be verified")]
+    Unverified,
+    #[error("failed to sign the invoice with the given key")]
+    SigningFailed,
+    #[error("key is corrupt")]
+    CorruptKey,
+    #[error("signature block is corrupt")]
+    CorruptSignature,
+    #[error("unknown signing key")]
+    UnknownSigningKey,
+    #[error("none of the signatures are made with a known key")]
+    NoKnownKey,
+}
+
+/// The role of a signer in a signature block.
+///
+/// Signatories on a signature must have an associated role, as defined in the
+/// specification.
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub enum SignatureRole {
+    #[serde(rename = "creator")]
```

Comment:
Nit: You could use the [serde `rename_all = "lowercase"`](https://serde.rs/container-attrs.html#rename_all) attribute here on the struct instead of individually renaming each one

Reply:
I didn't realize that worked on Enums!